src/mainloop.c

```diff
@@ -69,12 +71,63 @@ static int sock_set_non_blocking(int fd)
     return 0;
 }
 
-/* TODO: use command line options to specify */
-#define PORT 8081
 #define WEBROOT "./www"
+#define DEFAULT_PORT 8081
+
+static int cmd_get_port(char *arg_port)
+{
+    long ret;
+    char *endptr;
+
+    ret = strtol(arg_port, &endptr, 10);
+    if ((errno == ERANGE && (ret == LONG_MAX || ret == LONG_MIN)) ||
+        (errno != 0 && ret == 0)) {
+        perror("strtol");
+        exit(EXIT_FAILURE);
+    }
+
+    if (endptr == arg_port) {
+        fprintf(stderr, "No digits were found\n");
+        exit(EXIT_FAILURE);
+    }
+
+    /* strtol successfully parse arg_port, and check boundary condition */
+    if (ret <= 0 || ret > 65535) {
+        ret = DEFAULT_PORT;
+    }
+    return ret;
+}
+
+struct runtime_conf {
+    unsigned short int port;
```

Comment:
You can safely use `int` since we don't tend to optimize here.
See https://jacobegner.blogspot.com/2019/11/unsigned-integers-are-dangerous.html